### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by The-Normal-One1**
### The following lists were done to connect with API:

- Used [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as my backend
- Used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make API calls
- Fetched my books from Bookstore API when the component has rendered with useEffect.
- Refactored my add book and remove book features to persist your changes in the server
- Modified my reducer logic to include HTTP requests to the Bookstore API

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,48 +1,97 @@
...
-      const result = state.listBooks.filter((book) => book.item_id !== bookId);
-      return {
-        listBooks: result,
-      };
+      state.listBooks = state.listBooks.filter(
+        (book) => book.item_id !== bookId,
+      );
+      // return {
+      //   listBooks: result,
+      // };
```
Comment (elisha2kyakpo1) [inline src/redux/books/booksSlice.js]: - [x] Let's remove the commented code in order to make our codebase clean and readable.

File: src/components/books/bookAdd.js [hunk 2]
```diff
@@ -1,36 +1,31 @@
...
-        }),
-      );
+      const bookData = {
+        item_id: uuidv4(),
+        title,
+        author,
+      };
+      dispatch(addBook(bookData));
+      dispatch(postBook(bookData));
       resetInput();
```
Comment (elisha2kyakpo1) [inline src/components/books/bookAdd.js:30]: - [x] I was able to add and delete a book, but when I refreshed the page, I did not see my remaining books. I opened the dev tools to see the network tab, that's when I noticed a 400 error for a post request. Kindly take a look at the image below and make the changes.

book

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @The-Normal-One1, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description look excellent and professional.
✔️ No linter errors.
✔️ Descriptive readme file.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — APPROVED** [review]
### Hi  @The-Normal-One1 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**